scripts/pool_candidates.py

```
@@ -0,0 +1,100 @@
+    """rerank 상위 후보 (chunk_id·제목·점수) — retrieval_adapter 경로 미러 + 메타 노출."""
+    settings = get_settings()
+    qvec = await get_embedder().embed_query(query)
+    hits = await dense_search(qvec, top_k, domain=domain, settings=settings)
+    if not hits and domain:
+        hits = await dense_search(qvec, top_k, domain=None, settings=settings)
+    candidates = [(p.payload.get("content", ""), p.payload or {}) for p in hits]
+    try:
+        reranked = await anyio.to_thread.run_sync(get_reranker().rerank, query, candidates)
+        ranked = [(apply_metadata_weight(score, payload, settings), payload) for score, payload in reranked]
+        ranked.sort(key=lambda item: item[0], reverse=True)
+    except Exception:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)